api/api_test_suite.go

```diff
@@ -0,0 +1,206 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/cloudflare/cfssl/csr"
+	"github.com/cloudflare/cfssl/helpers/testsuite"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+)
+
+var (
+	keyRequest = csr.BasicKeyRequest{
+		A: "rsa",
+		S: 2048,
+	}
+	CAConfig = csr.CAConfig{
+		PathLength: 1,
+		Expiry:     "1/1/2016",
+	}
+	baseRequest = csr.CertificateRequest{
+		CN: "example.com",
+		Names: []csr.Name{
+			{
+				C:  "US",
+				ST: "California",
+				L:  "San Francisco",
+				O:  "Internet Widgets, LLC",
+				OU: "Certificate Authority",
+			},
+		},
+		Hosts:      []string{"ca.example.com"},
+		KeyRequest: &keyRequest,
+	}
+	CARequest = csr.CertificateRequest{
+		CN: "example.com",
+		Names: []csr.Name{
+			{
+				C:  "US",
+				ST: "California",
+				L:  "San Francisco",
+				O:  "Internet Widgets, LLC",
+				OU: "Certificate Authority",
+			},
+		},
+		Hosts:      []string{"ca.example.com"},
+		KeyRequest: &keyRequest,
+		CA:         &CAConfig,
+	}
+)
+
+func main() {
+	/* START UP CFSSL SERVER */
+	// We will test on this address and port. Be sure that these are free or
+	// the test will fail.
+	addressToTest := "127.0.0.1"
+	portToTest := 1234
+	// Set up a test server using our CA certificate and key.
+	CACert, CAKey, err := testsuite.CreateSelfSignedCert(CARequest)
+	if err != nil {
+		panic(err)
+	}
+	// Set up a test server using our CA certificate and key.
+	serverData := testsuite.CFSSLServerData{CA: CACert, CAKey: CAKey}
+	server, err := testsuite.StartCFSSLServer(addressToTest, portToTest, serverData)
+	if err != nil {
+		//panic(err)
+		//server.Kill()
+	}
+
+	/* COLLECT API OUTPUT */
+	var jsonStr = []byte(`{"domain": "example.com", "flavor": "ubiquitous"}`)
+	req, err := http.NewRequest("POST", "http://127.0.0.1:1234/api/v1/cfssl/bundle", bytes.NewBuffer(jsonStr))
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		panic(err)
+	}
+	defer resp.Body.Close()
+
+	fmt.Println("\n\nAPI OUTPUT:\n\n")
+	//fmt.Println("response Status:", resp.Status)
+	//fmt.Println("response Headers:", resp.Header)
+	body, _ := ioutil.ReadAll(resp.Body)
+	fmt.Println("response Body:", string(body))
+
+	/* COLLECT COMMAND LINE OUTPUT */
+	// Now make a request of our server and check that no error occurred.
+	// First we need a request to send to our server. We marshall the request
+	// into JSON format and write it to a temporary file.
+	jsonBytes, err := json.Marshal(baseRequest)
+	if err != nil {
+		panic(err)
+	}
+	tempFile, err := createTempFile(jsonBytes)
+	if err != nil {
+		os.Remove(tempFile)
+		panic(err)
+	}
+
+	// Now we make the request and check the output.
+	remoteServerString := "-remote=" + addressToTest + ":" + strconv.Itoa(portToTest)
```

Comment:
`fmt.Sprintf` can be used to make this a bit cleaner:

    remoteServerSTring := fmt.Sprintf("-remote=%s:%s", addressToTest, strconv.Itoa(portToTest))